controller/aks-cluster-config-handler.go

```diff
@@ -0,0 +1,890 @@
+	client.Authorizer = authorizer
+
+	return &client, nil
+}
+
+func newServiceClient(authorizer autorest.Authorizer, spec aksv1.AKSClusterConfigSpec) (*containersv209.ContainerServicesClient, error) {
+	if authorizer == nil {
+		return nil, fmt.Errorf("authorizer must not be nil")
+	}
+
+	baseURL := spec.BaseURL
+	if baseURL == "" {
+		baseURL = azure.PublicCloud.ResourceManagerEndpoint
+	}
+
+	client := containersv209.NewContainerServicesClientWithBaseURI(baseURL, spec.SubscriptionID)
+	client.Authorizer = authorizer
+
+	return &client, nil
+}
+
+// createOrUpdateCluster creates a new managed Kubernetes cluster
+func createOrUpdateCluster(ctx context.Context, secretsCache wranglerv1.SecretCache, clusterClient *containersv211.ManagedClustersClient, spec aksv1.AKSClusterConfigSpec) (c containersv211.ManagedCluster, err error) {
+
+	dnsPrefix := spec.DNSPrefix
+	if dnsPrefix == nil {
+		dnsPrefix = to.StringPtr(spec.ClusterName)
+	}
+
+	tags := make(map[string]*string)
+	for key, val := range spec.Tags {
+		if val != "" {
+			tags[key] = to.StringPtr(val)
+		}
+	}
+	displayName := spec.DisplayName
+	if displayName == "" {
+		displayName = spec.ClusterName
+	}
+	tags["displayName"] = to.StringPtr(displayName)
+
+	kubernetesVersion := spec.KubernetesVersion
+
+	var vmNetSubnetID *string
+	networkProfile := &containersv211.NetworkProfile{}
+	if hasCustomVirtualNetwork(spec) {
+		virtualNetworkResourceGroup := spec.ResourceGroup
+
+		//if virtual network resource group is set, use it, otherwise assume it is the same as the cluster
+		if spec.VirtualNetworkResourceGroup != "" {
+			virtualNetworkResourceGroup = spec.VirtualNetworkResourceGroup
+		}
+
+		vmNetSubnetID = to.StringPtr(fmt.Sprintf(
+			"/subscriptions/%v/resourceGroups/%v/providers/Microsoft.Network/virtualNetworks/%v/subnets/%v",
+			spec.SubscriptionID,
+			virtualNetworkResourceGroup,
+			spec.VirtualNetwork,
+			spec.Subnet,
+		))
+
+		networkProfile.DNSServiceIP = to.StringPtr(spec.NetworkDNSServiceIP)
+		networkProfile.DockerBridgeCidr = to.StringPtr(spec.NetworkDockerBridgeCIDR)
+		networkProfile.ServiceCidr = to.StringPtr(spec.NetworkServiceCIDR)
+
+		if spec.NetworkPlugin == "" {
+			networkProfile.NetworkPlugin = containersv211.Kubenet
+		} else {
+			networkProfile.NetworkPlugin = containersv211.NetworkPlugin(spec.NetworkPlugin)
+		}
+
+		// if network plugin is 'kubenet', set PodCIDR
+		if networkProfile.NetworkPlugin == containersv211.Azure {
+			networkProfile.PodCidr = to.StringPtr(spec.NetworkPodCIDR)
+		}
+
+		if spec.LoadBalancerSku != "" {
+			loadBalancerSku := containersv211.LoadBalancerSku(spec.LoadBalancerSku)
+			networkProfile.LoadBalancerSku = loadBalancerSku
+		}
+
+		if spec.NetworkPolicy != "" {
+			networkProfile.NetworkPolicy = containersv211.NetworkPolicy(spec.NetworkPolicy)
+		}
+	}
+
+	var agentPoolProfiles []containersv211.ManagedClusterAgentPoolProfile
+	if hasAgentPoolProfile(spec) {
```

Comment:
Does this mean "if user node pools" else "system only node pools" ?

Reply:
Yes, if `AKSClusterConfig` will not have any NodePools configured, default `system` NodePool will be used